rclcpp/include/rclcpp/subscription_options.hpp

```diff
@@ -75,6 +75,14 @@ struct SubscriptionOptionsBase
   TopicStatisticsOptions topic_stats_options;
 
   QosOverridingOptions qos_overriding_options;
+
+  /// Set callback group, parameter idiom style.
+  SubscriptionOptionsBase &
+  set_callback_group(rclcpp::CallbackGroup::SharedPtr group)
```

Comment:
If we only add one method, this is not being of much help.
It would be gread to add methods enabling chaining of all options.